x/orders/internal/keeper/genesis.go

```diff
@@ -0,0 +1,61 @@
+package keeper
+
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/dfinance/dnode/x/orders/internal/types"
+)
+
+// InitGenesis inits module genesis state: creates currencies.
+func (k Keeper) InitGenesis(ctx sdk.Context, data json.RawMessage) {
+	k.modulePerms.AutoCheck(types.PermRead)
+
+	state := types.GenesisState{}
+	k.cdc.MustUnmarshalJSON(data, &state)
+	for _, order := range state.Orders {
+		k.set(ctx, order)
+	}
+
+	if err := state.Validate(); err != nil {
```

Comment:
Better to check that Market exists here, otherwise it would lead to errors in the orderbook module